inc/functions/files.php

```diff
@@ -656,38 +662,28 @@ function rocket_clean_home( $lang = '' ) {
 	*/
 	do_action( 'before_rocket_clean_home', $root, $lang ); // phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals
 
-	// Delete homepage.
-	$files = glob( $root . '/{index,index-*}.{html,html_gzip}', GLOB_BRACE | GLOB_NOSORT );
-	if ( $files ) {
-		foreach ( $files as $file ) { // no array map to use @.
-			rocket_direct_filesystem()->delete( $file );
-		}
-	}
-
-	// Delete homepage pagination.
-	$dirs = glob( $root . '*/' . $GLOBALS['wp_rewrite']->pagination_base, GLOB_NOSORT );
-	if ( $dirs ) {
-		foreach ( $dirs as $dir ) {
-			rocket_rrmdir( $dir );
+	foreach ( _rocket_get_cache_dirs( $parse_url['host'], $cache_path ) as $dir ) {
+		$domain_entry = $dir . $parse_url['path'];
+		if ( ! $filesystem->exists( $domain_entry ) ) {
+			continue;
 		}
-	}
-
-	// Remove the hidden empty file for mobile detection on NGINX with the Rocket NGINX configuration.
-	$nginx_mobile_detect_files = glob( $root . '/.mobile-active', GLOB_BRACE | GLOB_NOSORT );
-	if ( $nginx_mobile_detect_files ) {
-		foreach ( $nginx_mobile_detect_files as $nginx_mobile_detect_file ) { // no array map to use @.
-			rocket_direct_filesystem()->delete( $nginx_mobile_detect_file );
+		$iterator = new DirectoryIterator( $domain_entry );
+
+		// Delete homepage.
+		// Remove the hidden empty file for mobile detection on NGINX with the Rocket NGINX configuration.
+		// Remove the hidden empty file for webp.
+		foreach ( $iterator as $domain_entry_item ) {
+			if ( $domain_entry_item->isFile() ) {
+				$filesystem->delete( $domain_entry_item->getPathname() );
+			}
 		}
-	}
 
-	// Remove the hidden empty file for webp.
-	$nowebp_detect_files = glob( $root . '/.no-webp', GLOB_BRACE | GLOB_NOSORT );
-	if ( $nowebp_detect_files ) {
-		foreach ( $nowebp_detect_files as $nowebp_detect_file ) { // no array map to use @.
-			rocket_direct_filesystem()->delete( $nowebp_detect_file );
+		// Delete homepage pagination.
+		$pagination_dir = $domain_entry . DIRECTORY_SEPARATOR . $GLOBALS['wp_rewrite']->pagination_base;
```

Comment:
I'm thinking we should use a `global $wp_rewrite` at the start of the function, and then use the following here:

```suggestion
		$pagination_dir = $domain_entry . DIRECTORY_SEPARATOR . $wp_rewrite->pagination_base;
```